common/src/main/java/dev/ftb/mods/ftbquests/block/QuestBarrierBlock.java

```diff
@@ -69,14 +80,26 @@ public RenderShape getRenderShape(BlockState state) {
 	@Override
 	public VoxelShape getCollisionShape(BlockState state, BlockGetter bg, BlockPos pos, CollisionContext ctx) {
 		if (EntityHooks.fromCollision(ctx) instanceof Player player
-				&& bg.getBlockEntity(pos) instanceof BarrierBlockEntity barrier
+				&& bg.getBlockEntity(pos) instanceof BaseBarrierBlockEntity barrier
 				&& barrier.isOpen(player)) {
 			return Shapes.empty();
 		}
 
 		return super.getCollisionShape(state, bg, pos, ctx);
 	}
 
+	@Override
+	protected VoxelShape getShape(BlockState blockState, BlockGetter bg, BlockPos pos, CollisionContext ctx) {
+		if (EntityHooks.fromCollision(ctx) instanceof Player player
+				&& bg.getBlockEntity(pos) instanceof BaseBarrierBlockEntity barrier
+				&& barrier.isOpen(player)
+				&& !barrier.hasPermissionToEdit(player)) {
```

Comment:
This concerns me a bit... Get shape can be called a lot IIRC... My concern is someone building a wall of these buggers then complaining about lag...

Reply:
Yeah, good point. I think it might be possible just to use the blockstate property here, let me do some experimentation...